ergotree-ir/src/sigma_protocol/dlog_group.rs

```diff
@@ -105,6 +107,20 @@ pub fn random_scalar_in_group_range() -> Scalar {
     Scalar::generate_vartime(&mut OsRng)
 }
 
+/// Attempts to create Scalar from BigUint
+/// Returns None if not in the range [0, modulus).
+pub fn from_biguint(b: BigUint) -> Option<Scalar> {
+    let bytes_be = b.to_bytes_be();
+    let bytes = bytes_be.as_slice().clone();
+
+    let mut bytes_32 = [0; 32];
+    for (i, v) in bytes.clone().iter().enumerate() {
+        bytes_32[i] = *v;
```

Comment:
`bytes` can be larger than 32 bytes since `BigUint` is not bounded.

EDIT: I mean although `BigInt` in ErgoScript is 256-bit we cannot presume that input always fit in 256 bits.